vterm.el

```diff
@@ -490,26 +492,27 @@ Argument EVENT process event."
     (run-hook-with-args 'vterm-exit-functions
                         (if (buffer-live-p buf) buf nil))))
 
-(defun vterm--window-size-change-26 (frame)
-  "Callback triggered by a size change of the FRAME.
-
-This is only used, when variable `emacs-version' < 27. Calls
-`vterm--window-size-change' for every window of FRAME."
-  (dolist (window (window-list frame))
-    (vterm--window-size-change window)))
-
-(defun vterm--window-size-change (window)
-  "Callback triggered by a size change of the WINDOW.
-
-Feeds the size change to the virtual terminal."
-  (with-current-buffer (window-buffer window)
-    (when (and (processp vterm--process)
-               (process-live-p vterm--process))
-      (let ((height (window-body-height window))
-            (width (window-body-width window))
-            (inhibit-read-only t))
-        (set-process-window-size vterm--process height width)
-        (vterm--set-size vterm--term height width)))))
+(defun vterm--window-adjust-process-window-size (process windows)
+  "Adjust process window size considering the width of line number."
+  (let* ((size (funcall window-adjust-process-window-size-function
+                        process windows))
+         (width (car size))
+         (height (cdr size))
+         (inhibit-read-only t))
+    (setq width (- width (vterm--get-margin-width)))
+    (when (and (processp process)
+               (process-live-p process))
+      (vterm--set-size vterm--term height width))
+    (cons width height)))
+
+(defun vterm--get-margin-width ()
+  "Get margin width of vterm buffer when `display-line-numbers-mode' is enabled."
+  (let ((width 0))
+    (when (and (boundp 'display-line-numbers)
+               display-line-numbers)
+      (require 'display-line-numbers)
```

Comment:
This should not be needed, since `display-line-numbers` is bound and set here.

Reply:
I think this is needed.

 `display-line-numbers` is defined in C from emacs26.1.

1.  I think we should still support emacs <26.1.
2. One can just enable display line numbers by `(setq display-line-numbers t)` in a mode hook without require display-line-numbers.el